src/sv-pipeline/04_variant_resolution/scripts/SR_genotype.opt_part2.sh

```diff
@@ -25,188 +125,175 @@ rare_both=$(awk '{if($1=="rare_both") print $2}' $metric_file)
 
 ##Require both##
 zcat $SR_counts|awk '{if ($NF>0) print $1"@"$3}'|sort|uniq -c|awk '{if ($1==2) print $2}'>two.sided.pass.just1read.txt
+# Output columns:
+# 1) VID@Sample (with at least 1 SR on both sides)
 
 join -j 2 <(cut -d"@" -f1 two.sided.pass.txt|sort|uniq -c) \
    <(cut -d"@" -f1 two.sided.pass.just1read.txt|sort|uniq -c) \
-   |awk '{print $0"\t" $2/$3}' \
-   >recover.bothsides.txt
-
+   | awk '{print $0"\t" $2/$3}' \
+   > recover.bothsides.txt
+# Output columns:
+# 1) VID (with at least 1 sample having bothside support)
+# 2) Number of samples with bothside support
+# 3) Number of samples with non-zero counts on both sides
+# 4) Fraction of samples with non-zero SR count that have sufficient counts on both sides, i.e. (2)/(3)
 
 ## Find passing variants based off ROC check##
-
 awk -v var=$rare_max  -v var1=$rare_single '{if ($NF>=var1 && $2<=var) print $1}' recover.txt>single.pass.txt
 awk -v var=$rare_max -v var1=$rare_both '{if ($NF>=var1  && $2<=var) print $1}' recover.bothsides.txt>both.pass.txt
 
 awk -v var=$common_min  -v var1=$common_single '{if ($NF>=var1 && $2>=var) print $1}' recover.txt>>single.pass.txt
+# Output columns:
+# 1) VID (one-sided variants that don't have too many background samples,
+#         i.e. samples with non-zero SR that are genotyped as hom-ref)
 awk -v var=$common_min -v var1=$common_both '{if ($NF>=var1  && $2>=var) print $1}' recover.bothsides.txt>>both.pass.txt
+# Output columns:
+# 1) VID (two-sided variants that don't have too many background samples,
+#         i.e. samples with non-zero SR that have two-sided support)
+
+# Note the VIDs in single.pass.txt and both.pass.txt are not mutually exclusive
 
 ##SR background variant failures##
-if [ $(cat both.pass.txt single.pass.txt \
-  |fgrep -wvf - int.bed \
-  |awk '{print $4}' \
-  |sort -u \
-  |fgrep -wf pass.srtest.txt \
-  |fgrep -wf <(cat recover.txt recover.bothsides.txt|awk '{print $1}'|sort -u)|wc -l) -gt 0 ]
-then
- cat both.pass.txt single.pass.txt \
-  |{ fgrep -wvf - int.bed || [[ $? == 1 ]]; } \
-  |awk '{print $4}' \
-  |sort -u \
-  |{ fgrep -wf pass.srtest.txt || [[ $? == 1 ]]; } \
-  |{ fgrep -wf <(cat recover.txt recover.bothsides.txt|awk '{print $1}'|sort -u) || [[ $? == 1 ]]; } \
-   >background.variant.fail.txt
-else 
- echo "">background.variant.fail.txt 
-fi
+cat both.pass.txt single.pass.txt \
+  | awk -F'\t' -v OFS='\t' 'ARGIND==1{keys[$1]; next} { if ($4 != "name" && !($4 in keys)) print }' - int.bed \
+  | awk '{print $4}' \
+  | sort -u \
+  | awk -F'\t' -v OFS='\t' 'ARGIND==1{keys[$1]; next} { if ($1 in keys) print }' pass.srtest.txt - \
+  | awk -F'\t' -v OFS='\t' 'ARGIND==1{keys[$1]; next} { if ($1 in keys) print }' <(cat recover.txt recover.bothsides.txt|awk '{print $1}'|sort -u) - \
+  > background.variant.fail.txt
 
 ##Pull out variants to genotype##
 join -j 1 -a 1 -a 2  -e "." -o 1.1 2.1 1.2 1.3 1.4 1.5 2.2 2.3 2.4 2.5 \
   <(zcat sr.geno.final.txt.gz \
-  |fgrep -wf both.pass.txt \
```

Comment:
Does this mean that we used to null out SR GTs if the SR background was too high but we are no longer doing so?

Reply:
This has more to do with two-sided support than SR background. 

It is a bit complicated, but basically there are two models for genotyping: a two-sided and a one-sided. Before, it would select the model based on whether the site was deemed one- or two-sided. The methodology for flagging a site as two-sided was fine for the purposes of just assigning a two-side support status to the variant (by checking what fraction of non-ref samples have two-sided support, above). 

However, the method also used to genotype according to the site-level both-side support status. It turns out that within a single site, there can be a mixture of two-side support among samples, and the cutoff method was not good enough. This wasn't an issue before since it happened that the one- and two-sided models were actually identical. But now that the one-sided model has different cutoffs, this approach was causing havoc in a lot of variants.

Here I have just removed the requirement that two-sided genotypes have two-sided status at the site level.